### PR TITLE
Add pptx support

### PR DESCRIPTION
**Pull request by tdonia**
one more bug to crack

### CODE REVIEW

File: coffee/docxTemplater.coffee [hunk 1]
```diff
@@ -2,10 +2,11 @@ XmlTemplater=require('./xmlTemplater')
 xmlUtil=require('./xmlUtil')
 
 DocXTemplater = class DocXTemplater extends XmlTemplater
+	tagByFileType = {pptx: "a:t", docx: "w:t"}
 	constructor:(content="",options={}) ->
 		super("",options)
 		@currentClass=DocXTemplater
-		@tagXml='w:t'
+		@tagXml = tagByFileType[options.fileType] ||= tagByFileType["docx"]
```
Comment (smitchsmith) [inline coffee/docxTemplater.coffee]: i'd put the second part in parens (just for clarity)

File: coffee/docxgen.coffee [hunk 2]
```diff
@@ -23,8 +24,11 @@ module.exports=class DocxGen
 		@qrCode= if @options.qrCode? then @options.qrCode else off
 		if @qrCode==true then @qrCode=DocUtils.unsecureQrCode
 		if @options.parser? then @parser=options.parser
+		@fileType = @fileTypeFromPath(@options.path) if @options.path != "" and @options.path != undefined
```
Comment (smitchsmith) [inline coffee/docxgen.coffee]: I think using coffee's `?` operator is more idiomatic - as opposed to `!= undefined`. it doesn't check for empty strings, though. so:

`@options.path? and @options.path != ""`

File: coffee/docxgen.coffee [hunk 3]
```diff
@@ -104,7 +128,8 @@ module.exports=class DocxGen
 				#Be aware that data-uri doesn't work for too big files: More Info http://stackoverflow.com/questions/17082286/getting-max-data-uri-size-in-javascript
 				document.location.href= "data:application/vnd.openxmlformats-officedocument.wordprocessingml.document;base64,#{result}"
 		result
-	getFullText:(path="word/document.xml") ->
+	getFullText:(path="") ->
+		path = @baseDoc if path == "" or path == undefined
```
Comment (smitchsmith) [inline coffee/docxgen.coffee]: same w/ the `?`

**Comment by tdonia** [comment]
ok - bugs fixed and PR feedback from @smitchsmith incorporated.  @mjlewis - anything you see?

**Comment by tdonia** [comment]
and so, it turns out active dev was on this branch: https://github.com/edi9999/docxtemplater/tree/1.x - not master.  so it'd be nice to redo this against that branch / take into account his PR feedback.  But for now, this will work / get us to the multipage etc issues that are the next big blocker.

**Pull request closed without merging**